packages/core/base/src/wallet.ts

```diff
@@ -0,0 +1,13 @@
+import { WalletAdapter } from './adapter';
+import { MessageSignerWalletAdapter, SignerWalletAdapter } from './signer';
+
+export type WalletName = string & { __brand__: 'WalletName' };
```

Comment:
This is a bit of a trick to get nominal types: https://medium.com/@KevinBGreene/surviving-the-typescript-ecosystem-branding-and-type-tagging-6cf6e516523d

The idea here is that any wallet can declare that its name is a `WalletName` by casting it. Downstream of this, the React/Vue/whatever components still have strong(ish) typing so that any string isn't accepted to select a wallet. Userland code never needs to cast wallet names, and they still get (some) type safety.

Reply:
Interesting, makes sense! I might suggest using the `Brand` generic that the article explains and adding a link to the article in a comment. That might help other confused people in the future :)

Reply:
Done.